src/main/java/com/github/princesslana/eriscasper/rx/websocket/RxWebSocket.java

```diff
@@ -29,31 +31,63 @@ public RxWebSocket(OkHttpClient http) {
             em -> {
               Request rq = new Request.Builder().url(url).build();
 
-              ws = http.newWebSocket(rq, new Listener(em));
+              ws = http.newWebSocket(rq, new Listener(RxWebSocket.this, em));
             })
+        .takeUntil(event -> closed)
```

Comment:
Referring to mutable things outside of the reactive stream is a bit of a code smell - it's kind of like the global variables of rx.

Reply:
I think I can remove that, I think those are in there for no reason at this point since I added in the basic level of `takeUntil`

Reply:
We should detect closing by seeing the Closed websocket event: https://github.com/princesslana/ErisCasper.java/blob/master/src/main/java/com/github/princesslana/eriscasper/rx/websocket/RxWebSocketEvent.java#L14

Reply:
gotcha, implemented that